puppetboard/docker_settings.py

```diff
@@ -1,4 +1,34 @@
 import os
+import tempfile
+import base64
+import binascii
+
+
+def cert_to_file(cert_file_or_string):
+    """
+    cert_to_file takes in a string, if it looks like a certificate, save the contents to a temporary
+    file and return the path to that temporary file
+
+    cert_to_file also supports base64 encoded certificates, in cases where newlines cause problems
+
+    Note: NamedTemporaryFile does not work great with Windows, but since this is for Docker
+    hopefully we'll be fine.
+    """
+    if isinstance(cert_file_or_string, str):
+        try:
+            cert_str = base64.b64decode(cert_file_or_string).decode()
+        except (UnicodeDecodeError, binascii.Error):
+            cert_str = cert_file_or_string
+
+        if '-----BEGIN' in cert_str:
+            with tempfile.NamedTemporaryFile(delete=False, suffix='puppetboard_cert') as tmpfile:
+                tmpfile.write(cert_str.encode())
+                tmpfile.close()
+            return tmpfile.name
+        else:
```

Comment:
since you return on 27, you can drop this else and untab line 29, as that return of `cert_file_or_string` will just be hit if we aren't dealing with a `-----BEGIN` cert. 

side note, whenever I have an if/else with a large chunk of code between the two clauses, it can be a good time for an in-line comment saying `#  we know the user passed us a file name; lets return it directly`

Reply:
> since you return on 27, you can drop this else and untab line 29, as that return of `cert_file_or_string` will just be hit if we aren't dealing with a `-----BEGIN` cert.

True, but IMHO the current code is perfectly readable.

> side note, whenever I have an if/else with a large chunk of code between the two clauses, it can be a good time for an in-line comment saying `# we know the user passed us a file name; lets return it directly`

+1 to that!